modules: hostap: fix DPP build error

fengming-ye opened the pull request:
Fix DPP build error when HOSTAPD enabled and DPP disabled.
Guard hapd_dpp_dispatch in both CONFIG_WIFI_NM_WPA_SUPPLICANT_DPP
and CONFIG_WIFI_NM_HOSTAPD_AP.

fengming-ye commented:
to fix issue https://github.com/zephyrproject-rtos/zephyr/issues/79897

jukkar commented:
Please add relevant information why the old commit needs to be reverted to the commit message.

fengming-ye commented:
> Please add relevant information why the old commit needs to be reverted to the commit message.

updated

jukkar commented:
But why do we need to revert the commit 1da74ef705bcf1723a3271266622948251253c0a, what kind of issue it is causing now?

fengming-ye commented:
> But wouldn't the original issue then be present what the [1da74ef](https://github.com/zephyrproject-rtos/zephyr/commit/1da74ef705bcf1723a3271266622948251253c0a) is fixing if we revert it?

If hostapd disables DPP (by disabling CONFIG_WIFI_NM_WPA_SUPPLICANT_DPP, which is zephyr only way to disable DPP),
L2 DPP code will be still there and can be tested, no build error, which is from original review comments request.
Only hostap will return fail when DPP L2 api is called, which won't be trouble.
So I don't see what 1da74ef is trying to fix. Could you pls explain it in detail?

krish2718 commented:
> > But wouldn't the original issue then be present what the [1da74ef](https://github.com/zephyrproject-rtos/zephyr/commit/1da74ef705bcf1723a3271266622948251253c0a) is fixing if we revert it?
> 
> If hostap disables DPP (by disabling CONFIG_WIFI_NM_WPA_SUPPLICANT_DPP, which is zephyr only way to disable DPP), L2 DPP code will be still there and can be tested, no build error, which is from original review comments request. Only hostap will return fail when DPP L2 api is called, which won't be trouble. So I don't see what [1da74ef](https://github.com/zephyrproject-rtos/zephyr/commit/1da74ef705bcf1723a3271266622948251253c0a) is trying to fix. Could you pls explain it in detail?

Please see https://github.com/zephyrproject-rtos/zephyr/issues/79897#issuecomment-2415994308 just including the header file is enough to get this error, but for some reason not seen with upstream.

fengming-ye committed:
modules: hostap: fix DPP build error

Fix DPP build error when HOSTAPD enabled and DPP disabled.
Guard hapd_dpp_dispatch in both CONFIG_WIFI_NM_WPA_SUPPLICANT_DPP
and CONFIG_WIFI_NM_HOSTAPD_AP.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

fengming-ye commented:
@jukkar @krish2718 updated fix. thanks for clarification.

krish2718 commented:
Not sure about twister policies in upstream, but should we add these combinations to avoid such issues?

jukkar commented:
> Not sure about twister policies in upstream, but should we add these combinations to avoid such issues?

That would be a good idea to add various config combinations to tests, not sure if `tests/net/wifi/wifi_nm` is a good place for those. Perhaps separate directory next to it to compile test various config combinations or enable-all type config.

krish2718 commented:
> Perhaps separate directory next to it to compile test various config combinations or enable-all type config

Just saw a PR merged with `build_all` that builds all drivers, we can have something similar for WPA supp?

jukkar commented:
> Just saw a PR merged with `build_all` that builds all drivers, we can have something similar for WPA supp?

Yeah we could, I can prepare a PR for that.
